openpype/pipeline/file_collection.py

```diff
@@ -0,0 +1,292 @@
+        Any[str, None]: Formattable string or None if not possible
+                        to convert
+    """
+    new_filename = None
+
+    if "#" in filename:
+        # use regex to convert #### to {:0>4}
+        def replace(match):
+            return "{{:0>{}}}".format(len(match.group()))
+        new_filename = re.sub("#+", replace, filename)
+
+    elif "%" in filename:
+        # use regex to convert %04d to {:0>4}
+        padding = re.search("%(\\d)+d", filename)
+        padding = padding.group(1) if padding else 1
+        new_filename = re.sub(
+            "%.*d",
+            "{{:0>{}}}".format(padding),
+            filename
+        )
+
+    return new_filename
+
+
+def get_publishing_representation(
+    instance,
+    file_path,
+    frame_start=None,
+    frame_end=None,
+    log=None,
+    only_existing=False,
+    reviewable=False,
+):
+    """Get representation with expected files.
+
+    Args:
+        instance (pyblish.Instance): instance
+        file_path (str): file path
+        frame_start (Optional[int]): first frame
+        frame_end (Optional[int]): last frame
+        log (Optional[Logger]): logger
+        only_existing (Optional[bool]): Ensure that files exists.
+        reviewable (Optional[bool]): reviewable
+
+
+    Returns:
+        dict[str, Any]: representation
+    """
+    log = log or Logger.get_logger(__name__)
+    file_ext = os.path.splitext(file_path)[-1].lstrip(".")
+    output_dir = os.path.dirname(file_path)
+    # get families from instance and add family from data
+    families = set(
+        instance.data.get("families", []) + [instance.data["family"]]
+    )
+
+    tags = []
+    if reviewable:
+        tags.append("review")
+
+    # prepare base representation data
+    representation = {
+        "name": file_ext,
+        "ext": file_ext,
+        "stagingDir": output_dir,
+        # QUESTION: should we use persistent staging dir always?
+        "stagingDir_persistent": True,
+        "tags": tags,
+    }
+
+    # add frame range data
+    if frame_start is not None:
+        representation["frameStart"] = frame_start
+    if frame_end is not None:
+        representation["frameEnd"] = frame_end
+
+    # collect files from sequential path or single file in list
+    collected_file_frames = \
+        collect_basenames_from_sequential_path(
+            file_path, frame_start, frame_end, only_existing)
+
+    # set slate frame
+    if (
+        "slate" in families
+        # make sure frame range is set
+        and _add_slate_frame_to_collected_frames(
+            collected_file_frames, frame_start, frame_end)
+    ):
+        log.info("Slate frame added to collected frames.")
+        representation["frameStart"] = frame_start - 1
+
+    if len(collected_file_frames) == 1:
+        representation["files"] = collected_file_frames.pop()
+    else:
+        representation["files"] = collected_file_frames
+
+    return representation
+
+
+def _add_slate_frame_to_collected_frames(
```

Comment:
Would this be more understandable if the wording was `_prepend_slate_frame_...` so that you know where it adds it?